build: hack around unnecessary docker image pulls on circleci

petermattis opened the pull request:
Before `TestAdminLossOfQuorum` was taking ~45-50s. After it is taking ~6-7s.

Fixes #5849.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5854)

<!-- Reviewable:end -->

petermattis committed:
build: hack around unnecessary docker image pulls on circleci

Fixes #5849.

tamird commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[acceptance/cluster/docker.go, line 82 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5SNiK4ClVSVoMDJD:-KEY5SNiK4ClVSVoMDJE:96393908) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L82)):*
"tag" is surprising, perhaps s/tag/repoTag/

---

*[acceptance/cluster/docker.go, line 91 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5P8PSobfIRlKPpYI:-KEY5P8PSobfIRlKPpYJ:-415155578) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L91)):*
needs more words

---

*[acceptance/cluster/docker.go, line 109 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5wuFgF6wzb2j0ACg:-KEY5wuGZmM4U-tsBxG9:302666376) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L109)):*
is this for the logging? needs a comment, and perhaps the return value should be logged as well.

---

*[acceptance/cluster/docker.go, line 459 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY59fnqBzfgDXmjIuc:-KEY59fnqBzfgDXmjIud:698164617) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L459)):*
can you follow the style guide here? https://github.com/cockroachdb/cockroach/blob/master/CONTRIBUTING.md#style-guide

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-:-KEY63jUUfL8gMWlVjBi:1417088721)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

*[acceptance/cluster/docker.go, line 82 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5SNiK4ClVSVoMDJD:-KEY72d7Ku9vtwxrsCUX:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L82)):*
Done.

---

*[acceptance/cluster/docker.go, line 91 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5P8PSobfIRlKPpYI:-KEY74GyFZfzFThQ3oWG:2058150792) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L91)):*
Added `image ` as a prefix. But it is pretty obvious what this is. Note that the log message shouldn't happen in practice as the image will be cached.

---

*[acceptance/cluster/docker.go, line 109 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5wuFgF6wzb2j0ACg:-KEY7P7yJFKB_Skmzhbv:-1722901619) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L109)):*
It was for logging when I was listing images. No longer necessary as it will never fire. Removed.

---

*[acceptance/cluster/docker.go, line 459 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY59fnqBzfgDXmjIuc:-KEY6gaFt4d28udW7PWS:1563690079) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L459)):*
Where did that come from? Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/cluster/docker.go, line 91 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5P8PSobfIRlKPpYI:-KEY7spQONXWUXK167bg:-1468162757) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L91)):*
ah, this is logged between other log lines that give context. still, i'd have followed other examples here and prefixed with `ImageList: <name of image>`

---

*[acceptance/cluster/docker.go, line 109 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5wuFgF6wzb2j0ACg:-KEY7xocmAXezL0SuESs:851230772) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L109)):*
you don't need the wrapping `func` anymore, unless you're trying to late-bind the arguments to the logger.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854)*
<!-- Sent from Reviewable.io -->

petermattis commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[acceptance/cluster/docker.go, line 91 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5P8PSobfIRlKPpYI:-KEY8cZEaOQ2OLjnjKoO:-1621351685) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L91)):*
Changed to `ImageList`.

---

*[acceptance/cluster/docker.go, line 109 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854#-KEY5wuFgF6wzb2j0ACg:-KEY8ezkw3NPgW1F-sbJ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/8c66212208b2e97a28fc4303526197a6b983ca7f/acceptance/cluster/docker.go#L109)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/5854)*
<!-- Sent from Reviewable.io -->
